modyn/selector/internal/selector_strategies/abstract_selection_strategy.py

```diff
@@ -404,7 +404,7 @@ def get_available_labels(self) -> list[int]:
                 database.session.query(SelectorStateMetadata.label)
                 .filter(
                     SelectorStateMetadata.pipeline_id == self._pipeline_id,
-                    SelectorStateMetadata.seen_in_trigger_id >= self._next_trigger_id - self.tail_triggers
+                    SelectorStateMetadata.seen_in_trigger_id >= self._next_trigger_id - self.tail_triggers - 1
```

Comment:
This query is done **after** the trigger (so `_next_trigger_id` has already been incremented) and thus we need the `-1` .  In other queries (for presampling strategies) there is no `-1` since they are performed before incrementing the counter

Reply:
just so I understand: `get_available_labels` should return all labels until the last trigger, correct? so if we have

add labels A,B,C -> trigger -> add label D -> get all labels

we should return A,B,C correct?

in the integration test you fixed, I see you added the trigger, but don't we also need to inform about a new data point (label D here) to ensure it is doing exactly that?

Reply:
Thank you! This LGTM